samplers/src/main/java/io/opentelemetry/contrib/sampler/LinksBasedSampler.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.sampler;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import java.util.List;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A Sampler that uses the sampled flag of the span links if present. If at least one span link is
+ * sampled, then this span will be sampled. Otherwise, it is not sampled. If the span has no span
+ * links, this Sampler will use the "root" sampler that it is built with.
+ */
+@Immutable
+final class LinksBasedSampler implements Sampler {
```

Comment:
This class needs to be public. 

Also, do we need the `@Immutable` annotation?